modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/directives/AuthDirectives.scala

```diff
@@ -0,0 +1,78 @@
+package ch.epfl.bluebrain.nexus.kg.service.directives
+
+import akka.http.scaladsl.model.headers.OAuth2BearerToken
+import akka.http.scaladsl.server.Directives.{authorizeAsync, extractCredentials, provide}
+import akka.http.scaladsl.server.directives.FutureDirectives.onComplete
+import akka.http.scaladsl.server.directives.RouteDirectives.reject
+import akka.http.scaladsl.server.{AuthorizationFailedRejection, Directive0, Directive1}
+import cats.Show
+import cats.syntax.show._
+import ch.epfl.bluebrain.nexus.commons.http.UnexpectedUnsuccessfulHttpResponse
+import ch.epfl.bluebrain.nexus.commons.iam.IamClient
+import ch.epfl.bluebrain.nexus.commons.iam.acls.{Path, Permission}
+import ch.epfl.bluebrain.nexus.commons.iam.identity.Caller
+import ch.epfl.bluebrain.nexus.commons.types.HttpRejection.UnauthorizedAccess
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Success
+
+trait AuthDirectives {
+
+  private val prefix = Path("kg")
+
+  /**
+    * Checks if the caller has permissions ''perms'' on the path ''resource''.
+    *
+    * @param resource the path which the caller wants to access
+    * @param perm     the permission to be checked
+    */
+  def authorizeResource(resource: Path, perm: Permission)(implicit iamClient: IamClient[Future],
+                                                          caller: Caller,
+                                                          ec: ExecutionContext): Directive0 =
+
+    authorizeAsync {
+      iamClient
+        .getAcls(resource)
+        .map(_.toMap.values.find(!_.contains(perm)).isEmpty)
```

Comment:
create a private def like so:

```
private def permissions(acl: AccessControlList): Permissions = acl.acl.foldLeft(Permissions.empty) {
    case (acc, ac) => acc ++ ac.permissions
  }
```
such that you can just do:
```
iamClient.getAcls(resource).map(acl => permissions(acl).contains(perm))
```

Reply:
Ideally we should have this in the `AccessControlList` class along with possibly a `contains` method.